### PR TITLE
Mattg/use firebase sdk

### PR DESCRIPTION
**Pull request by mattgrosso**
Deving with a copy of your much larger database has, once again, raised the fear that this whole thing might accidentally cost a bunch of money if I'm not careful. I've been using Firebase via web addresses and api endpoints but the way they really want us to work with it is through their SDK (software development kit). So, I looked into it and it does see like it could save us a lot of downloads (which is the thing that's been adding up).

I also went back and refactored the styles for the grid layout in the header. It should now gracefully scale as users get more movies rated.

### CODE REVIEW

File: src/components/Header.vue [hunk 1]
```diff
@@ -103,92 +90,118 @@ export default {
...
-    randomPoster () {
+    randomBanner () {
       const rand = Math.floor(Math.random() * this.$store.getters.allMoviesAsArray.length);
 
       if (this.$store.getters.allMoviesAsArray[rand]) {
-        return `https://image.tmdb.org/t/p/original${this.$store.getters.allMoviesAsArray[rand].movie.poster_path}`;
+        return `https://image.tmdb.org/t/p/original${this.$store.getters.allMoviesAsArray[rand].movie.backdrop_path}`;
       } else {
-        return "https://live.staticflickr.com/65535/47980547206_7a02cb7f3f_h.jpg";
+        return "https://image.tmdb.org/t/p/original/1TvNazsE9WvRIxxeZkvL7IVVgzD.jpg";
```
Comment (rkaufman13) [inline src/components/Header.vue:99]: guessing this doesn't scale with users D:

File: src/store/index.js [hunk 2]
```diff
@@ -14,6 +16,13 @@ const sortByVoteCount = (a, b) => {
   return 0;
 }
 
+// Firebase
+const firebaseConfig = {
+  databaseURL: "https://movie-log-8c4d5-default-rtdb.firebaseio.com",
```
Comment (rkaufman13) [inline src/store/index.js:21]: i might just be being paranoid but i'd store this url as an env variable if possible (i think surge does allow this, never done it myself)

**Review by rkaufman13 — APPROVED** [review]
Couple teensy nitpicks (actually one is just a question), and I admit the frontend is fairly beyond me, but I ran your branch locally and everything seems to work. (I can't get the "re-rate" link to work but it doesn't work off main either so I'm assuming that's still in progress.) Good enough for me if it's good enough for you.

**Review by bgoegan — APPROVED** [review]

File: src/components/Header.vue (hunk 1 above)
Comment (mattgrosso) [reply src/components/Header.vue:99]: Sure it does. That's the banner from Jurassic Park. That's universal.

File: src/store/index.js (hunk 2 above)
Comment (mattgrosso) [reply src/store/index.js:21]: Well, this URL isn't meant to be secret. Anyone with a browser can monitor the network tab and get our address. 

The way to make the DB secure would be to set up some actual auth rules on Firebase. But that's kind of hard so I haven't done it.

**Pull request merged**